uaclient/entitlements/base.py

```diff
@@ -589,12 +589,16 @@ def _disable_dependent_services(self):
 
         return True
 
+    def _check_for_reboot(self) -> bool:
+        """Check if system needs to be rebooted."""
+        return util.should_reboot()
+
     def _check_for_reboot_msg(self, operation: str) -> None:
         """Check if user should be alerted that a reboot must be performed.
 
         @param operation: The operation being executed.
         """
-        if util.should_reboot():
+        if self._check_for_reboot():
             print(
                 status.MESSAGE_ENABLE_REBOOT_REQUIRED_TMPL.format(
                     operation=operation
```

Comment:
Do we need this in the base entitlement?
It could suggest that is a reboot is required for livepatch to finish enabling, which I don't think makes sense, right?

Reply:
No, because on the base entitlement we don't pass the packages that were installed to the `should_reboot` function

Reply:
Yes, for non-repo entitlements, that will happen. But this is what is happening today. Do you think we should pass an empty list for those situations ?

Reply:
I think passing an empty list is the same as not passing a list in the current implementation.

But it is probably okay to leave it as is.